mpl_interactions/generic.py

```diff
@@ -578,7 +579,14 @@ def hyperslicer(
     controls
     """
 
-    arr = np.asarray(np.squeeze(arr))
+    arr = np.squeeze(arr)
+
+    arr_type = "numpy"
+    if "xarray.core.dataarray.DataArray" in str(arr.__class__):
+        arr_type = "xarray"
+    elif "dask.array.core.Array" in str(arr.__class__):
+        arr_type = "dask"
```

Comment:
completeness mostly? and I wrote that before I tried it with a dask array. I feel like its fine to have just because if anything comes up that we need to special case for dask?